language/tools/move-unit-test/src/test_runner.rs

```diff
@@ -149,19 +208,27 @@ impl SharedTestingConfig {
             .unwrap();
         };
 
+        let stackless_model = if self.use_stackless_vm {
+            let model =
+                run_model_builder_with_compilation_flags(&self.source_files, &[], Flags::testing())
+                    .unwrap_or_else(|e| panic!("Unable to build stackless bytecode: {}", e));
+            Some(model)
+        } else {
+            None
+        };
+
         for (function_name, test_info) in &test_plan.tests {
-            let move_vm = MoveVM::new();
-            let mut session = move_vm.new_session(&self.starting_storage_state);
-            let log_context = NoContextLog::new();
-
-            match session.execute_function(
-                &test_plan.module_id,
-                &IdentStr::new(function_name).unwrap(),
-                vec![], // no ty args, at least for now
-                serialize_values(test_info.arguments.iter()),
-                &mut GasStatus::new(&self.cost_table, GasUnits::new(self.execution_bound)),
-                &log_context,
-            ) {
+            let exec_result = if self.use_stackless_vm {
+                self.execute_via_stackless_vm(
+                    stackless_model.as_ref().unwrap(),
+                    test_plan,
+                    function_name,
+                    test_info,
+                )
+            } else {
+                self.execute_via_move_vm(test_plan, function_name, test_info)
```

Comment:
Is there any way of getting the final storage state out of this after execution? Eventually we'll want to support printing out global states when a test fails. It's fine if it's not supported, but would be interesting if we could this if e.g., the stackless and bytecode tests differ.

Reply:
Yes, we can have the two `execute_via_*` functions return a tuple `(VMResult<Vec<Vec<u8>>>, InMemoryStorage)`, which is easy to do for both the Move VM version and the stackless VM version.

Reply:
Having it return a `ChangeSet` at the end would be the best if possible as that would fit in the most with my current implementation of this I'm working on, but I realize that it very well might not be. Either way, this isn't something that I think needs to be done in this PR and can be be done later on.